src/transformers/utils/hub.py

```diff
@@ -628,8 +626,58 @@ def _resumable_file_manager() -> "io.BufferedWriter":
     return cache_path
 
 
-def get_file_from_repo(
-    path_or_repo: Union[str, os.PathLike],
+def try_to_load_from_cache(cache_dir, repo_id, filename, revision=None):
+    """
+    Explores the cache to return the latest cached file for a given revision.
+    """
+    if revision is None:
+        revision = "main"
+
+    model_id = repo_id.replace("/", "--")
+    model_cache = os.path.join(cache_dir, f"models--{model_id}")
+    if not os.path.isdir(model_cache):
+        # No cache for this model
+        return None
+
+    # Resolve refs (for instance to convert main to the associated commit sha)
+    cached_refs = os.listdir(os.path.join(model_cache, "refs"))
+    if revision in cached_refs:
+        with open(os.path.join(model_cache, "refs", revision)) as f:
+            revision = f.read()
```

Comment:
(nit) why not try to access the file directly (as you're not using the full list)